pkg/edgeview/src/basics.go

```diff
@@ -0,0 +1,608 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	colorRED     = "\033[1;31m%s\033[0m"
+	colorBLUE    = "\033[1;34m%s\033[0m"
+	colorCYAN    = "\033[1;36m%s\033[0m"
+	colorGREEN   = "\033[0;32m%s\033[0m"
+	colorYELLOW  = "\033[0;93m%s\033[0m"
+	colorPURPLE  = "\033[1;95m%s\033[0m"
+	colorRESET   = "\033[0m"
+)
+
+var (
+	netopts       []string
+	pubsubopts    []string
+	pubsubpersist []string
+	pubsublarge   []string
+	sysopts       []string
+	logdirectory  []string
+)
+
+// all the supported options
+func initOpts() {
+	netopts = []string{
+		"acl",
+		"app",
+		"arp",
+		"connectivity",
+		"flow",
+		"if",
+		"mdns",
+		"nslookup",
+		"ping",
+		"route",
+		"socket",
+		"speed",
+		"tcp",
+		"tcpdump",
+		"trace",
+		"url",
+		"wireless"}
+
+	pubsubopts = []string{
+		"baseosmgr",
+		"domainmgr",
+		"downloader",
+		"edgeview",
+		"global",
+		"loguploader",
+		"newlogd",
+		"nim",
+		"nodeagent",
+		"tpmmgr",
+		"vaultmgr",
+		"volumemgr",
+		"watcher",
+		"zedagent",
+		"zedclient",
+		"zedmanager",
+		"zedrouter"}
+
+	pubsubpersist = []string{
+		"nim",
+		"tpmmgr",
+		"volumemgr",
+		"zedagent",
+		"zedclient",
+		"zedmanager",
+		"zedrouter"}
+
+	pubsublarge = []string{
+		"zedagent",
+		"zedmanager"}
+
+	sysopts = []string{
+		"app",
+		"configitem",
+		"cp",
+		"datastore",
+		"download",
+		"hw",
+		"lastreboot",
+		"model",
+		"newlog",
+		"pci",
+		"ps",
+		"cipher",
+		"usb",
+		"shell",
+		"techsupport",
+		"volume",
+	}
+
+	logdirectory = []string{
+		"/persist/newlog/keepSentQueue/",
+		"/persist/newlog/devUpload/",
+		"/persist/newlog/appUpload/",
+		"/persist/newlog/failedUpload/",
+	}
+}
+
+func getHostname() string {
+	var hostname string
+	retStr, err := runCmd("hostname", false, false)
+	if err != nil {
+		return hostname
+	}
+	hostname = strings.TrimSuffix(retStr, "\n")
```

Comment:
No need for this if you use os.Hostname()

Reply:
see above.